### ISSUE TITLE
[BUG] uriparser should decode pct-encoded

### ISSUE THREAD

**Issue by wakabayashik** [Haivision/srt#1173]
**Describe the bug**
uriparser should decode pct-encoded

**To Reproduce**
1. create an listener program with SRT-APIs (srt_listen, srt_accept...) and run as a listener.
2. run "srt-live-transmit srt://192.168.1.1:12345?streamid=%23!::u=admin,r=bluesbrothers1_hi file://con | NUL" as a caller
3. retrieve streamid by srt_getsockflag(s, SRTO_STREAMID, ...) on listener program.

**Expected behavior**
streamid should become "#!::u=admin,r=bluesbrothers1_hi",
but it is still "%23!::u=admin,r=bluesbrothers1_hi"

**Desktop (please provide the following information):**
 - OS: listener (Windows Server 2016)、caller (Windows 8.1)
 - SRT Version / commit ID: v1.4.1. (built it myself by vs2013 with some patches)

**Additional context**
https://tools.ietf.org/html/rfc3986
```
         foo://example.com:8042/over/there?name=ferret#nose
         \_/   \______________/\_________/ \_________/ \__/
          |           |            |            |        |
       scheme     authority       path        query   fragment
          |   _____________________|__
         / \ /                        \
         urn:example:animal:ferret:nose
```
because '#' is delimiter of fragment part, the query cannot contain '#' without pct-encoding.

 * unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"
 * sub-delims    = "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "="
 * pct-encoded = "%" HEXDIG HEXDIG
 * pchar = unreserved / pct-encoded / sub-delims / ":" / "@"
 * query = *( pchar / "/" / "?" )

**Comment by maxsharabayko** [comment]
More references on the topic: https://url.spec.whatwg.org/

**Comment by maxsharabayko** [comment]
Resolved by #2015.
